internal/ast/expression.go

```diff
@@ -86,69 +92,75 @@ func copyExpressionResult(src, dst *Expression) {
 	dst.ResultStringValue = src.ResultStringValue
 	dst.ResultSymbol = src.ResultSymbol
 	dst.ResultType = src.ResultType
+	dst.NativeZserioType = src.NativeZserioType
 }
 
-func evaluateExpressionType(typeRef *TypeReference, scope *Package) (ExpressionType, error) {
+// evaluateExpressionType returns the expression type and the bit
+// size of the underlying type for a type reference.
+// For example, if the type reference is an int16, ExpressionTypeInt, 16
+// is returned. For types where the bit count is not relevant, the bit
+// count is 0.
```

Comment:
The comment doesn't seem to match reality: you return the full TypeReference, not just the width of the type.

Reply:
Thank you, good catch! I indeed modified the function without checking the comment. I fixed that.